### PR TITLE
PWN-5477 jsonrpc server and client telemetry instrumentation

### PR DESCRIPTION
**Pull request by btolfa**
## Problem

We have to alter (add middleware) `jsonrpsee` server and client behavior - inject and extract correlation id between services.
Related to https://github.com/p2p-org/jsonrpsee/pull/1 
https://p2pvalidator.atlassian.net/browse/PWN-5477

## Checklist

- [ ] code changes are covered with tests
- [ ] README.md files are up to date
- [ ] CI runs without errors
- [ ] code review passed (at least one other person)

### CODE REVIEW

**Review by BorovkovSergey — APPROVED** [review]

File: src/telemetry.rs [hunk 1]
```diff
@@ -44,31 +52,63 @@ impl Telemetry {
...
+                    pipeline = pipeline.with_service_name(name.to_string());
+                }
+
+                pipeline = pipeline.with_trace_config(
+                    sdktrace::config()
+                        .with_resource(resource)
+                        .with_sampler(sdktrace::Sampler::AlwaysOn),
+                );
+
+                pipeline.install_batch(runtime::Tokio)?
```
Comment (kstep) [inline src/telemetry.rs:78]: These lines are the same for all cases. Can we move them out of the `match`? The problem is different types (`CollectorPipeline` vs. `AgentPipeline`). Maybe this macro will help?

```rust
        macro_rules! tracer {
            ($name:ident, $pipeline:expr) => {{
                let mut pipeline = $pipeline;
                if let Some(ref name) = $name {
                    pipeline = pipeline.with_service_name(name.to_string());
                }

                pipeline = pipeline.with_trace_config(
                    sdktrace::config()
                        .with_resource(resource)
                        .with_sampler(sdktrace::Sampler::AlwaysOn),
                );

                pipeline.install_batch(runtime::Tokio)?
            }};
        }
```

Comment (btolfa) [reply src/telemetry.rs]: Fixed

File: src/server.rs [hunk 2]
```diff
@@ -1,93 +1,104 @@
...
+    {
+        signal.await;
         match self.stop().await {
             Ok(_) => {
-                log::info!("server stopped successfully");
+                tracing::info!("server stopped successfully");
             },
             Err(error) => {
-                log::warn!("failed to stop the server: {error}");
+                tracing::warn!("failed to stop the server: {error}");
```
Comment (kstep) [inline src/server.rs:71]: We decided to use `warn!`/`info`/etc without prefixes before.

File: src/telemetry.rs [hunk 3]
```diff
@@ -44,32 +71,33 @@ impl Telemetry {
     /// We are using `impl Subscriber` as return type to avoid having to spell out the actual
     /// type of the returned subscriber, which is indeed quite complex.
     pub fn init(
-        name: String,
+        resource: Resource,
         tracing_settings: TracingSettings,
     ) -> anyhow::Result<(Self, impl Subscriber + Sync + Send)> {
         global::set_text_map_propagator(TraceContextPropagator::default());
 
+        let name = resource.get(semcov::resource::SERVICE_NAME);
```
Comment (kstep) [inline src/telemetry.rs:79]: Do you use `name` now? Maybe remove it.

Comment (btolfa) [reply src/telemetry.rs:79]: It used by formatter a few lines below. It should be created here, because `trace!` macro consumes `resource`;

**Pull request merged**